lib/render/Renderer.cpp

```diff
@@ -491,7 +491,8 @@ void Renderer::renderColorbar(){
 	if (makeColorbarTexture()) return;
 
     lgl->Color4fv(whitecolor);
-	
+
+    glActiveTexture( GL_TEXTURE0 );	
```

Comment:
Is this need because some other renderer is not leaving OpenGL in the default state? I.e. is there a _paintGL() method that returns without restoring the active texture to GL_TEXTURE0. If that is the case better to fix the renderer or this problem will crop up again elsewhere

Reply:
I believe this is the case.  Wouldn't it be prudent for all renderers to set their intended texture unit before drawing, and restore state after drawing?

Reply:
Yes that is the case, and we should strive to leave OpenGL in a default state, however I think that renderers should setup their own state within reason, and I think that setting the texture unit is reasonable.

Reply:
Unfortunately because of the hundreds of state flags OpenGL supports it is not practical to require each renderer to check the value of every state flag that might impact its rendering. This places too much of a burden on the renderer developer. That is why the policy has always been that a renderer developer can assume OpenGL state flags are in the default state on entry to paintGL() (with a couple of exceptions), and a renderer developer is responsible for restoring any changes to state made within paintGL() to their default values before returning from paintGL(). The default active texture unit is GL_TEXTURE0.

Given that this problem is associated with the isorenderer my guess is that it's @shaomeng code that is not cleaning up state. You might want to kick this back over to Sam

Reply:
Okey dokey.  @shaomeng, I'm kicking it over to you.